### DISCUSSION TITLE
How to localize path segments?

### DISCUSSION THREAD

**Discussion by openscript** [vikejs/vike#1633] [Discussion] [answered]
I have paths like `/docs/` or `/contact/` that I would like to localize. 

So the file system routing for `/docs/` has an alias `/doku/`. How would you do that?

**Comment by brillout** [comment]
I’m not sure I understand what you want to achieve.

**Reply by openscript** [reply]
The app will be available in 4 languages. I want to translate the urls:

- /en/residential/tv-subscription/channel-lists.html
- /de/privatkunden/tv-abo/senderliste.html
- /it/clienti-privati/abbonamento-tv/elenco-di-canali.html
- /fr/clients-prives/abonnement-tv/liste-des-chaines.html

How would you translate a segment like `residential` without having a folder for each in the file system routing?

**Reply by brillout** [reply, marked as answer]
I'd use `/pages/residential/tv-subscription/channel-lists/+route.js` with `onBeforeRoute()`.

**Reply by openscript** [reply]
Would this topic be a good contribution to the i18n page in the docs?

**Reply by brillout** [reply]
Yes :+1:
